pkg/deploy/config.go

```diff
@@ -48,7 +48,6 @@ type Configuration struct {
 	ARMAPICABundle                     *string                `json:"armApiCaBundle,omitempty"`
 	ARMAPIClientCertCommonName         *string                `json:"armApiClientCertCommonName,omitempty"`
 	ARMClientID                        *string                `json:"armClientId,omitempty"`
-	BillingE2EStorageAccountID         *string                `json:"billingE2EStorageAccountId,omitempty"`
 	BillingServicePrincipalID          *string                `json:"billingServicePrincipalId,omitempty"`
```

Comment:
where is BillingServicePrincipalID used?

Reply:
Indeed it is no longer being used and can also be removed. I investigated it last week.
I created a separate Jira issue for this, intending to remove it on a separate pull request.
Is it ok to remove `BillingServicePrincipalId` on a different pull request? 
Or would you prefer that I do it on this one?

Reply:
let's delete it in this PR

Reply:
To keep things simpler, I'm good with having one PR for billing e2e config and another PR for the unused billing service principal ID.

If you want to delete both in this PR, that's fine with me as well - I'll approve either.

Reply:
Just make sure it's clear in your PR description that you'll cleanup the other stuff in separate PRs.

Reply:
Removed `billingServicePrincipalId` on the new commits below. Also removed a role assignment that was being given to that `billingServicePrincipalId`, because it made sense to me that if we no longer have that service principal ID we no longer need to assign it any role. But is that conclusion correct? If the role assignment should be kept, please let me know.